### PR TITLE
Maintenance

### PR DESCRIPTION
**Pull request by veidenberg**
Non-functional cleanup/maintenance:
* Updated API spec to reflect the current endpoint implementation
* Updated type hints (for pyton 3.8+) and deprecated `.dict()` calls
* Removed unused imports and one endpoint
* Updated README

See related conversation in the [parent PR](https://github.com/Ensembl/ensembl-web-metadata-api/pull/42).

### CODE REVIEW

**Comment by kamaldodiya** [comment]
Since, the changes here are across all the endpoints I would locally check all the endpoint and put the response in the Description itself to ascertain that there is no regression.

File: README.md [hunk 1]
```diff
@@ -1,11 +1,17 @@
...
 ### Deploy the app and run docker-compose:
- 
- `$ git clone https://github.com/Ensembl/ensembl-web-metadata-api.git`
- `$  cd ensembl-web-metadata-api`
- `$  docker-compose -f docker-compose.yml up`
+```bash
+git clone https://github.com/Ensembl/ensembl-web-metadata-api
+cd ensembl-web-metadata-api
+vi .env #add values for PORT, GRPC_HOST, GRPC_PORT
+source .env
```
Comment (Mehrnaz-Charkhchi) [inline README.md:9]: We don't need this line here while working with docker containers.

**Comment by veidenberg** [comment]
> Since, the changes here are across all the endpoints I would locally check all the endpoint and put the response in the Description itself to ascertain that there is no regression.

I really don't think it's needed here. As I [mentioned in the parent PR](https://github.com/Ensembl/ensembl-web-metadata-api/pull/42#issuecomment-1944201908), there are no changes in the endpoints that could have any effect on the functionality or regression (besides the version bump in requirements.txt but that's already on main). Is there a particular code change you are concerned about?
That being said, we do need automatic regression/contract tests, as PR comment section is very ineffective for that (especially if we want to test all the endpoints and response types). For now, I ran few requests manually and these seemed to match the live instance.

**Comment by kamaldodiya** [comment]
>Is there a particular code change you are concerned about?

**Review by kamaldodiya — APPROVED** [review]

**Pull request merged**